regression/threaded_data_transfer/regression_large_data_async_transfers.c

```diff
@@ -0,0 +1,156 @@
+//===--- regression_large_data_async_transfers.c ------------------------------===//
+//
+// OpenMP API Version 4.5
+//
+// this tests performs asynchrous data transfer from the host to the device. this
+// test is mean to stress out the runtime by performing 8 asynchrous operations 
+// at once
+//
+// Routine being tested
+// omp_target_memcpy_async
+//
+// Author: Aaron Liu <olympus@udel.edu> Oct 2023
+////===----------------------------------------------------------------------===//
+
+#include <math.h>
+#include <omp.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#define N 32768
+#define number_of_threads 10000
+
+int main() {
+  int errors = 0;
+  int host = omp_get_initial_device();
+  int device = omp_get_default_device();
+
+  // allocate space on host & target
+  int array_size8 = N;
+  int array_size7 = (N * .875);
+  int array_size6 = (N * .750);
+  int array_size5 = (N * .635);
+  int array_size4 = (N * .500);
+  int array_size3 = (N * .375);
+  int array_size2 = (N * .250);
+  int array_size1 = (N * .125);
+
+  double *device_memory8 =
+      (double *)omp_target_alloc(sizeof(double) * array_size8, device);
+
+  double *host_memory8 = (double *)malloc(sizeof(double) * array_size8);
+  double *host_memory7 = (double *)malloc(sizeof(double) * array_size7);
+  double *host_memory6 = (double *)malloc(sizeof(double) * array_size6);
+  double *host_memory5 = (double *)malloc(sizeof(double) * array_size5);
+  double *host_memory4 = (double *)malloc(sizeof(double) * array_size4);
+  double *host_memory3 = (double *)malloc(sizeof(double) * array_size3);
+  double *host_memory2 = (double *)malloc(sizeof(double) * array_size2);
+  double *host_memory1 = (double *)malloc(sizeof(double) * array_size1);
+
+  for (int i; i < array_size4; i++) {
+    host_memory4[i] = i;
+    if (i < array_size3) {
+      host_memory3[i] = i;
+    }
+    if (i < array_size2) {
+      host_memory2[i] = i;
+    }
+    if (i < array_size1) {
+      host_memory1[i] = i;
+    }
+  }
+
+  /* copy to target */
+  omp_target_memcpy_async(device_memory8, host_memory8,
+                          sizeof(double) * array_size8, 0, 0, device, host, 0,
+                          NULL);
+
+#pragma omp target is_device_ptr(device_memory8) device(device)
```

Comment:
This target task is dependent on the `asynchronous omp_target_memcpy_async()` task; therefore, appropriate synchronization mechanism is needed (either creating an explicit dependency between the two tasks or adding `taskwait` directive before this target task.